<launch>
  <!-- board parameters -->
  <arg name="squaresX" default="7" />
  <arg name="squaresY" default="4" />
  <arg name="squareLength" default="0.036" />


  <!-- topic parameters -->
  <arg name="camera_info_topic" default="/kinect2/qhd/camera_info" />
  <arg name="camera_image_topic" default="/kinect2/qhd/image_color_rect" />
  <arg name="frame_name" default="kinect2_rgb_optical_frame" />

  <node pkg="pose_publisher" type="publish_pose_chessboard" name="board_pose" output="screen">
     <param name="squaresX" type="int" value="$(arg squaresX)"/>
     <param name="squaresY" type="int" value="$(arg squaresY)"/>
     <param name="squareLength" type="double" value="$(arg squareLength)"/>
     <param name="camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
     <param name="camera_image_topic" type="str" value="$(arg camera_image_topic)"/>
     <param name="frame_name" type="str" value="$(arg frame_name)"/>
  </node>
</launch>
